ext/storage_sources/s3_store/CMakeLists.txt

```diff
@@ -0,0 +1,82 @@
+project(s3_store CXX)    
+    
+include(ExternalProject)
+include(GNUInstallDirs)
+
+if (ENABLE_S3)
+    # Download and install the AWS CPP SDK into the build directory.
+    ExternalProject_Add(aws-sdk
```

Comment:
This is a good start, but I don't think using an `ExternalProject_Add` should be the default behaviour for building the aws s3 extension, since its imposing a requirement for the user to have both `git` installed and a network connection to build the sources (and of course adds a significant amount of build time). I think we should expose a configuration option that lets the end user choose to either:
1. Use a system installed aws sdk i.e. via `find_package`
2.  Fetch and build the project via an `ExternalProject_Add`
We can create quite easily do this via a `config_choice` variable:
```
config_choice(    
    IMPORT_S3_SDK    
    "Specifies how to import the S3 SDK"    
    OPTIONS    
    "none;IMPORT_S3_SDK_NONE;"    
    "package;IMPORT_S3_SDK_PACKAGE;HAVE_S3_SDK_PACKAGE"    
    "external;IMPORT_S3_SDK_EXTERNAL;"    
)    
```
then have an if-else depending on the config value (generalized):
```
if(IMPORT_S3_SDK_NONE)
   return()
elseif(IMPORT_S3_SDK_PACKAGE)
   find_package(AWSSDK)
   ....
else()
   ExternalProject_Add(...)
   ...
endif()
```

Reply:
Thanks Alison, have created [WT-8705](https://jira.mongodb.org/browse/WT-8705) for adding this configuration option!

Reply:
Ah ok, If you don't intend to make the change in this PR, we need to at least update `cmake/third_party/aws_sdk.cmake` to return early before calling `ExternalProject_Add` if `ENABLE_S3` is not enabled. 
I feel really strongly that `ExternalProject` should _never_ be called without explicit configuration consent by the user. Otherwise people will be checking out WiredTiger and building the aws-sdk regardless if they actually need it. Also the `ENABLE_S3` doc string should make it clear we will be fetching the aws sdk.